File: Python_TRR/4.py
# Viết bằng Python
# Bài 4: Tính giai thừa bằng đệ quy.

def factorial(n): # Đệ quy là một kỹ thuật lập trình mà một hàm gọi chính nó.
    if n == 0 or n == 1: # Điều kiện dừng của đệ quy.
        return 1 # Giai thừa của 0 và 1 là 1.
    return n * factorial(n - 1) # Gọi đệ quy để tính giai thừa của n.

# Ví dụ
n = 5
print(f"Giai thừa của {n} là: {factorial(n)}") 

# Kết quả
# Giai thừa của 5 là: 120

File: C_TRR/4.c
// Viết bằng C
// Bài 4: Tính giai thừa bằng đệ quy.

#include <stdio.h>

long long factorial(int n) 
{
    if (n == 0 || n == 1)
        return 1;
    return n * factorial(n - 1);
}

int main() 
{
    int n = 5;
    printf("Giai thừa của %d là: %lld\n", n, factorial(n));
    return 0;
}

